app/main.py

```diff
@@ -1,3 +1,60 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        return Distance(
+            self.km
+            + (other.km if isinstance(other, Distance) else other))
+
+    def __iadd__(self,
+                 other: Distance | float | int
+                 ) -> Distance | float | int:
+        self.km = (self.km
+                   + (other.km if isinstance(other, Distance) else other)
+                   )
+        return self
+
+    def __mul__(self,
+                other: float | int
+                ) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self,
+                    other: float | int
+                    ) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self,
+               other: Distance | float | int
+               ) -> bool:
+        return self.km < (other.km if isinstance(other, Distance) else other)
+
+    def __gt__(self,
+               other: Distance | float | int
+               ) -> bool:
+        return self.km > (other.km if isinstance(other, Distance) else other)
+
+    def __eq__(self,
+               other: Distance | float | int
+               ) -> bool:
+        return self.km == (other.km if isinstance(other, Distance) else other)
+
+    def __le__(self,
+               other: Distance | float | int
+               ) -> bool:
+        return self.km <= (other.km if isinstance(other, Distance) else other)
```

Comment:
DRY - don't repeat yourself 

 `(other.km if isinstance(other, Distance) else other)` - create a separate method for this